tests/components/mobile_app/test_init.py

```diff
@@ -93,15 +111,24 @@ async def _test_create_cloud_hook(
         patch("homeassistant.components.cloud.async_is_connected", return_value=True),
         patch(
             "homeassistant.components.cloud.async_get_or_create_cloudhook",
-            autospec=True,
+            side_effect=mock_get_or_create_cloudhook,
         ) as mock_async_get_or_create_cloudhook,
+        patch(
+            "homeassistant.components.cloud.async_listen_cloudhook_change",
+            side_effect=mock_listen_cloudhook_change,
+        ),
     ):
-        cloud_hook = "https://hook-url"
-        mock_async_get_or_create_cloudhook.return_value = cloud_hook
-
         assert await hass.config_entries.async_setup(config_entry.entry_id)
         await hass.async_block_till_done()
         assert config_entry.state is ConfigEntryState.LOADED
+
+        assert cloudhook_change_callback is not None
+
+        # Store the callback for use in additional_steps
+        hass.data.setdefault("_test", {})["cloudhook_change_callback"] = (
```

Comment:
We shouldn't store test functions in `hass.data`. Break this out to a fixture that can be used and give access to the callback in a test or return the callback from this function.